src/Packetery/Module/Order/PacketAutoSubmitter.php

```diff
@@ -0,0 +1,171 @@
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Module\Options\Provider $optionsProvider Options provider.
+	 * @param Repository              $orderRepository Order repository.
+	 * @param PacketSubmitter         $packetSubmitter Packet submitter.
+	 * @param Module\Options\Page     $optionsPage Options page.
+	 * @param Core\Validator\Order    $orderValidator Order validator.
+	 */
+	public function __construct(
+		Module\Options\Provider $optionsProvider,
+		Repository $orderRepository,
+		PacketSubmitter $packetSubmitter,
+		Module\Options\Page $optionsPage,
+		Core\Validator\Order $orderValidator
+	) {
+		$this->optionsProvider = $optionsProvider;
+		$this->orderRepository = $orderRepository;
+		$this->packetSubmitter = $packetSubmitter;
+		$this->optionsPage     = $optionsPage;
+		$this->orderValidator  = $orderValidator;
+	}
+
+	/**
+	 * Registers listeners.
+	 *
+	 * @return void
+	 */
+	public function register(): void {
+		if ( false === $this->optionsProvider->isPacketAutoSubmissionEnabled() ) {
+			return;
+		}
+
+		$mappedEvents = $this->optionsProvider->getPacketAutoSubmissionMappedUniqueEvents();
+		foreach ( $mappedEvents as $mappedEvent ) {
+			if ( self::EVENT_ON_ORDER_COMPLETED === $mappedEvent ) {
+				add_action(
+					'woocommerce_order_status_completed',
+					function ( int $orderId, \WC_Order $wcOrder ) {
+						$this->handleEvent( self::EVENT_ON_ORDER_COMPLETED, $wcOrder );
+					},
+					10,
+					2
+				);
+				continue;
+			}
+
+			if ( self::EVENT_ON_ORDER_PROCESSING === $mappedEvent ) {
+				add_action(
+					'woocommerce_order_status_processing',
+					function ( int $orderId, \WC_Order $wcOrder ) {
+						$this->handleEvent( self::EVENT_ON_ORDER_PROCESSING, $wcOrder );
+					},
+					10,
+					2
+				);
+			}
+		}
+	}
+
+	/**
+	 * Handle event.
+	 *
+	 * @param string    $event Event.
+	 * @param \WC_Order $wcOrder WC Order.
+	 *
+	 * @return void
+	 */
+	public function handleEvent( string $event, \WC_Order $wcOrder ): void {
+		if ( self::EVENT_NONE === $event ) {
+			return;
+		}
+
+		if ( false === $this->optionsProvider->isPacketAutoSubmissionEnabled() ) {
+			return;
+		}
+
+		$order = $this->orderRepository->getByWcOrder( $wcOrder );
+		if ( null === $order || null !== $order->getPacketId() ) {
+			return;
+		}
+
+		if ( false === $this->orderValidator->validateFinalWeight( $order ) ) {
+			return;
+		}
+
+		$carrier = $order->getCarrier();
+		if ( null !== $carrier && $carrier->requiresSize() && false === $this->orderValidator->validateSize( $order ) ) {
+			return;
+		}
+
+		$paymentGateway = wc_get_payment_gateway_by_order( $wcOrder );
+		if ( false === $this->optionsPage->hasPaymentGateway( $paymentGateway ) ) {
+			return;
+		}
+
+		$mappingEventForGateway = $this->optionsProvider->getPacketAutoSubmissionEvenForPaymentGateway( $paymentGateway->id );
+		if ( self::EVENT_NONE === $mappingEventForGateway || $mappingEventForGateway !== $event ) {
+			return;
+		}
+
+		$resultsCounter = [
```

Comment:
Tuhle inicializaci máme v modulu už 3x, co to zavést třeba jako statickou property submitteru?

Reply:
Vzhledem k tomu, že pole se přenáší odkazem, tak bych to viděl spíše na jednoduchou továrničku.